File: source/1502. Can Make Arithmetic Progression From Sequence/main.py
class Solution:
    def canMakeArithmeticProgression(self, arr: list[int]) -> bool:
        arr.sort()
        idx = 1
        length = len(arr)
        diff = arr[1] - arr[0]
        while idx < length-1:
            if arr[idx+1] - arr[idx] != diff: return False
            idx += 1
        del idx, length, diff
        return True

# https://leetcode.com/problems/can-make-arithmetic-progression-from-sequence/solutions/6590083/solution-for-can-make-arithmetic-progres-qj7q/

File: source/1502. Can Make Arithmetic Progression From Sequence/main.cpp
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    bool canMakeArithmeticProgression(vector<int>& arr) {
        sort(arr.begin(), arr.end(), [](int a, int b)
             { return a > b; });

        int difference = arr[0] - arr[1];
        for (int idx = 1; idx < (arr.size() - 1); idx++) {
            if (arr[idx] - arr[idx + 1] != difference) {
                return false;
            }
        }

        return true;
    }
};